Source/compositorclient/CMakeLists.txt

```diff
@@ -33,10 +33,34 @@ set(PLUGIN_COMPOSITOR_IMPLEMENTATION "None" CACHE STRING "Defines which implemen
 
 find_package(CompileSettingsDebug CONFIG REQUIRED)
 find_package(${NAMESPACE}Core REQUIRED)
+find_package(${NAMESPACE}COM REQUIRED)
+
+add_library(${TARGET} SHARED Module.cpp)
+
+target_link_libraries(${TARGET}
+    PRIVATE
+        CompileSettingsDebug::CompileSettingsDebug
+        ${NAMESPACE}Core::${NAMESPACE}Core
+        ${NAMESPACE}VirtualInput::${NAMESPACE}VirtualInput
+        ${NAMESPACE}COM::${NAMESPACE}COM
+)
+
+set_target_properties(${TARGET}
+    PROPERTIES
+        CXX_STANDARD 11
+        CXX_STANDARD_REQUIRED YES
+        VERSION ${PROJECT_VERSION}
+        SOVERSION ${PROJECT_VERSION_MAJOR}
+)
+
+target_include_directories(${TARGET}
+    PUBLIC
+        $<INSTALL_INTERFACE:include/${NAMESPACE}>
+)
 
 if("${PLUGIN_COMPOSITOR_IMPLEMENTATION}" STREQUAL "Nexus")
     if (COMPOSITORCLIENT_IMPLEMENTATION_PATH)
-        add_subdirectory(${COMPOSITORCLIENT_IMPLEMENTATION_PATH})
+        include(${COMPOSITORCLIENT_IMPLEMENTATION_PATH}/CMakeLists.txt)
```

Comment:
Why did you change ```add_subdirectory``` to ```include```?  

I think its highly uncommon to use ```include``` to include a sub directory that contains a CMakeLists.txt. Unless you have a technical reason of doing this, please use ```add_subdirectory``` to add a entry point for the build system generator . Use ```include``` if you need functionality implemented in a specific cmake file.

Reply:
@bramoosterhuis 
Following is the issue with add_subdirectory():
As the add_library() is called in the top level CMakeLists.txt, it doesn't allow target_link_libraries() in the implementation folder CMakeLists.txt.